metrics/scaling/k8s_scale_rapid.sh

```diff
@@ -0,0 +1,299 @@
+
+	for reqs in $(seq ${STEP} ${STEP} ${NUM_PODS}); do
+		info "Testing replicas ${reqs} of ${NUM_PODS}"
+		# Generate the next yaml file
+
+		local runtime_command
+		if [ -n "$RUNTIME" ]; then
+			runtime_command="s|@RUNTIMECLASS@|${RUNTIME}|g"
+		else
+			runtime_command="/@RUNTIMECLASS@/d"
+		fi
+
+		local input_template
+		local generated_file
+		if [ "$use_api" != "no" ]; then
+			input_template=$input_json
+			generated_file=$generated_json
+		else
+			input_template=$input_yaml
+			generated_file=$generated_yaml
+		fi
+
+		sed -e "s|@REPLICAS@|${reqs}|g" \
+			-e $runtime_command \
+			-e "s|@DEPLOYMENT@|${deployment}|g" \
+			-e "s|@LABEL@|${LABEL}|g" \
+			-e "s|@LABELVALUE@|${LABELVALUE}|g" \
+			-e "s|@GRACE@|${grace}|g" \
+			-e "s#@PODCOMMAND@#${pod_command}#g" \
+			< ${input_template} > ${generated_file}
+
+		info "Applying changes"
+		local start_time=$(date +%s%N)
+		if [ "$use_api" != "no" ]; then
+			# If this is the first launch of the deploy, we need to use a different URL form.
+			if [ $reqs == ${STEP} ]; then
+				curl -s ${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments -XPOST -H 'Content-Type: application/json' -d@${generated_file} > /dev/null
+			else
+				curl -s ${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments/${deployment} -XPATCH -H 'Content-Type:application/strategic-merge-patch+json' -d@${generated_file} > /dev/null
+			fi
+		else
+			kubectl apply -f ${generated_file}
+		fi
+
+		kubectl rollout status --timeout=${wait_time}s deployment/${deployment}
+		local end_time=$(date +%s%N)
+		local total_milliseconds=$(( (end_time - start_time) / 1000000 ))
+		info "Took $total_milliseconds ms ($end_time - $start_time)"
+
+		sleep ${settle_time}
+		grab_stats $total_milliseconds $reqs
+	done
+}
+
+cleanup() {
+	info "Cleaning up"
+
+	# First try to save any results we got
+	metrics_json_end_array "BootResults"
+
+	local start_time=$(date +%s%N)
+	kubectl delete deployment --wait=true --timeout=${delete_wait_time}s ${deployment} || true
+	for x in $(seq 1 ${delete_wait_time}); do
+		local npods=$(kubectl get pods -l=${LABEL}=${LABELVALUE} -o=name | wc -l)
+		if [ $npods -eq 0 ]; then
+			echo "All pods have terminated at cycle $x"
+			local alldied=true
+			break;
+		fi
+		sleep 1
+	done
+	local end_time=$(date +%s%N)
+	local total_milliseconds=$(( (end_time - start_time) / 1000000 ))
+	if [ -z "$alldied" ]; then
+		echo "ERROR: Not all pods died!"
+	fi
+	info "Delete Took $total_milliseconds ms ($end_time - $start_time)"
+
+	local json="$(cat << EOF
+	"Delete": {
+		"Result": ${total_milliseconds},
+		"Units" : "ms"
+	}
+EOF
+)"
+
+	metrics_json_add_fragment "$json"
+	metrics_json_save
+
+	cleanup_stats $delete_wait_time
+
+	k8s_api_shutdown
+}
+
+show_vars()
+{
+	echo -e "\nEnvironment variables:"
+	echo -e "\tName (default)"
+	echo -e "\t\tDescription"
+	echo -e "\tTEST_NAME (${TEST_NAME})"
+	echo -e "\t\tCan be set to over-ride the default JSON results filename"
```

Comment:
`TEST_NAME` is not supported to over-ride the default results filename, from the code it uses a default value (`k8s rapid`), so we can add a note saying it or remove it entirely.